--- freetype-2.1.10/src/autofit/aflatin.c.fixautofit	2005-08-24 15:34:14.000000000 +0200
+++ freetype-2.1.10/src/autofit/aflatin.c	2005-08-24 15:34:45.000000000 +0200
@@ -559,6 +559,8 @@
   af_latin_metrics_scale( AF_LatinMetrics  metrics,
                           AF_Scaler        scaler )
   {
+    metrics->root.scaler.render_mode = scaler->render_mode;
+	  
     af_latin_metrics_scale_dim( metrics, scaler, AF_DIMENSION_HORZ );
     af_latin_metrics_scale_dim( metrics, scaler, AF_DIMENSION_VERT );
   }
