diff -uNr Linux-PAM-0.75/libpamc/test/modules/pam_secret.c Linux-PAM-0.75/libpamc/test/modules/pam_secret.c
--- Linux-PAM-0.75/libpamc/test/modules/pam_secret.c	Sat Jan 20 17:29:47 2001
+++ Linux-PAM-0.75/libpamc/test/modules/pam_secret.c	Tue Jul 10 00:14:23 2001
@@ -206,6 +206,7 @@
     pwd = getpwnam(user);
     if ((pwd == NULL) || (pwd->pw_dir == NULL)) {
 	D(("user [%s] is not known", user));
+	return NULL;
     }
 
     length_id = strlen(pwd->pw_dir) + sizeof(SECRET_FILE_FORMAT);
