--- dvd+rw-tools-5.21.4.10.8/growisofs_mmc.cpp  2004-07-16 14:48:57.000000000 +0200
+++ dvd+rw-tools-5.21.4.10.8/growisofs_mmc.cpp	2005-04-09 18:16:09.000000000 +0200
@@ -1484,8 +1484,17 @@
 	cmd[1] = 0x01;		// "IMMED"
 	cmd[2] = mode;		// "Close session"
 	cmd[9] = 0;
-	if ((err=cmd.transport()))
-	    sperror ("CLOSE SESSION",err);
+	
+	// it seems, that pioneer is a bit crappy
+	while (err=cmd.transport()) {
+	    if (SK(err)==0x2 && ASC(err)==0x04 && ASCQ(err)==0x07) {
+		    sperror ("CLOSE SESSION (but try to continue)",err);
+		    usleep(10000);
+	    } else {
+		    sperror ("CLOSE SESSION",err);
+		    break;
+	    }
+	}
 
 	if (wait_for_unit (cmd)) break;
  
