--- source/Makefile.in.ORIG	2005-10-14 12:49:58.000000000 +0200
+++ source/Makefile.in	2005-10-14 13:02:31.000000000 +0200
@@ -25,6 +25,7 @@
 PYTHON=@PYTHON@
 PERL=@PERL@
 
+LAZYLDFLAGS=-Wl,-z,now
 TERMLDFLAGS=@TERMLDFLAGS@
 TERMLIBS=@TERMLIBS@
 PRINT_LIBS=@PRINT_LIBS@
@@ -71,6 +72,9 @@
 # The permissions to give the executables
 INSTALLPERMS = 0755
 
+# The permissions for setuid executables
+INSTALLSUPERMS = 4755
+
 # set these to where to find various files
 # These can be overridden by command line switches (see smbd(8))
 # or in smb.conf (see smb.conf(5))
@@ -132,6 +136,8 @@
 	bin/profiles@EXEEXT@ bin/ntlm_auth@EXEEXT@ \
 	bin/smbcquotas@EXEEXT@
 
+SUBIN_PROGS = @EXTRA_SUBIN_PROGS@
+
 TORTURE_PROGS = bin/smbtorture@EXEEXT@ bin/msgtest@EXEEXT@ \
 	bin/masktest@EXEEXT@ bin/locktest@EXEEXT@ \
 	bin/locktest2@EXEEXT@ bin/nsstest@EXEEXT@ bin/vfstest@EXEEXT@
@@ -558,6 +564,10 @@
 
 UMOUNT_OBJ = client/smbumount.o
 
+MCIFS_OBJ = client/mount.cifs.o
+
+UMCIFS_OBJ = client/umount.cifs.o
+
 NMBLOOKUP_OBJ = utils/nmblookup.o $(PARAM_OBJ) $(LIBNMB_OBJ) \
                $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) $(SECRETS_OBJ) $(LIBSAMBA_OBJ)
 
@@ -703,7 +713,7 @@
 ######################################################################
 # now the rules...
 ######################################################################
-all : SHOWFLAGS proto_exists $(SBIN_PROGS) $(BIN_PROGS) $(SHLIBS) \
+all : SHOWFLAGS proto_exists $(SBIN_PROGS) $(BIN_PROGS) $(SUBIN_PROGS) $(SHLIBS) \
 	$(MODULES) @EXTRA_ALL_TARGETS@
 
 pam_smbpass : SHOWFLAGS bin/pam_smbpass.@SHLIBEXT@
@@ -872,11 +882,19 @@
 
 bin/smbmnt@EXEEXT@: $(MNT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MNT_OBJ) $(DYNEXP) $(LDFLAGS) 
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
 
 bin/smbumount@EXEEXT@: $(UMOUNT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS)
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
+
+client/mount.cifs@EXEEXT@: $(MCIFS_OBJ) bin/.dummy
+	@echo Linking $@
+	@$(CC) $(FLAGS) -o $@ $(MCIFS_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
+
+client/umount.cifs@EXEEXT@: $(UMCIFS_OBJ) bin/.dummy
+	@echo Linking $@
+	@$(CC) $(FLAGS) -o $@ $(UMCIFS_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
 
 bin/testparm@EXEEXT@: $(TESTPARM_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
@@ -1312,7 +1330,7 @@
 
 bin/t_snprintf@EXEEXT@: lib/snprintf.c
 	$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(DYNEXP) -DTEST_SNPRINTF lib/snprintf.c -lm
-install: installbin installman installscripts installdat installswat installmodules @INSTALLCLIENT@
+install: installbin installsubin installman installscripts installdat installswat installmodules @INSTALLCLIENT@
 
 install-everything: install installmodules
 
@@ -1334,6 +1352,8 @@
 	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(SBINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(SBIN_PROGS)
 	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(BIN_PROGS)
 
+installsubin: all installdirs
+	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLSUPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(SUBIN_PROGS)
 
 # Some symlinks are required for the 'probing' of modules. 
 # This mechanism should go at some point..
