diff -ur k3b-0.12.10.orig/src/k3bsystemproblemdialog.cpp k3b-0.12.10/src/k3bsystemproblemdialog.cpp
--- k3b-0.12.10.orig/src/k3bsystemproblemdialog.cpp	2005-12-15 15:08:58.000000000 +0200
+++ k3b-0.12.10/src/k3bsystemproblemdialog.cpp	2005-12-22 02:01:04.000000000 +0200
@@ -190,36 +190,6 @@
 					 i18n("Install a more recent version of the cdrtools."),
 					 false ) );
     }
-    
-#ifdef Q_OS_LINUX
-
-    //
-    // Since kernel 2.6.8 older cdrecord versions are not able to use the SCSI subsystem when running suid root anymore
-    // So for we ignore the suid root issue with kernel >= 2.6.8 and cdrecord < 2.01.01a02
-    //
-    if( K3b::simpleKernelVersion() >= K3bVersion( 2, 6, 8 ) &&
-	k3bcore->externalBinManager()->binObject( "cdrecord" )->version < K3bVersion( 2, 1, 1, "a02" ) ) {
-      if( k3bcore->externalBinManager()->binObject( "cdrecord" )->hasFeature( "suidroot" ) )
-	problems.append( K3bSystemProblem( K3bSystemProblem::CRITICAL,
-					   i18n("%1 will be run with root privileges on kernel >= 2.6.8").arg("cdrecord <= 2.01.01a02"),
-					   i18n("Since Linux kernel 2.6.8 %1 will not work when run suid "
-						"root for security reasons anymore.").arg("cdrecord <= 2.01.01a02"),
-					   i18n("Use K3bSetup to solve this problem."),
-					   true ) );
-    }
-    else if( !k3bcore->externalBinManager()->binObject( "cdrecord" )->hasFeature( "suidroot" ) && getuid() != 0 ) // not root
-      problems.append( K3bSystemProblem( K3bSystemProblem::CRITICAL,
-					 i18n("%1 will be run without root privileges").arg("cdrecord"),
-					 i18n("It is highly recommended to configure cdrecord "
-					      "to run with root privileges. Only then cdrecord "
-					      "runs with high priority which increases the overall "
-					      "stability of the burning process. Apart from that "
-					      "it allows changing the size of the used burning buffer. "
-					      "A lot of user problems could be solved this way. This is also "
-					      "true when using SuSE's resmgr."),
-					 i18n("Use K3bSetup to solve this problem."),
-					 true ) );
-#endif
   }
 
   if( !k3bcore->externalBinManager()->foundBin( "cdrdao" ) ) {
@@ -229,18 +199,6 @@
 				       i18n("Install the cdrdao package."),
 				       false ) );
   }
-  else {
-#ifdef Q_OS_LINUX
-    if( !k3bcore->externalBinManager()->binObject( "cdrdao" )->hasFeature( "suidroot" ) && getuid() != 0 )
-      problems.append( K3bSystemProblem( K3bSystemProblem::CRITICAL,
-					 i18n("%1 will be run without root privileges").arg("cdrdao"),
-					 i18n("It is highly recommended to configure cdrdao "
-					      "to run with root privileges to increase the "
-					      "overall stability of the burning process."),
-					 i18n("Use K3bSetup to solve this problem."),
-					 true ) );
-#endif
-  }
 
   if( !k3bcore->deviceManager()->dvdWriter().isEmpty() ) {
     if( !k3bcore->externalBinManager()->foundBin( "growisofs" ) ) {
