--- source/configure.in.ORIG	2005-10-14 12:49:53.000000000 +0200
+++ source/configure.in	2005-10-14 12:56:42.000000000 +0200
@@ -233,6 +233,7 @@
 AC_SUBST(SMBWRAP_OBJS)
 AC_SUBST(SMBWRAP_INC)
 AC_SUBST(EXTRA_BIN_PROGS)
+AC_SUBST(EXTRA_SUBIN_PROGS)
 AC_SUBST(EXTRA_SBIN_PROGS)
 AC_SUBST(EXTRA_ALL_TARGETS)
 
@@ -3326,7 +3327,8 @@
 	*linux*)
 		AC_MSG_RESULT(yes)
 		AC_DEFINE(WITH_SMBMOUNT,1,[Whether to build smbmount])
-		EXTRA_BIN_PROGS="$EXTRA_BIN_PROGS bin/smbmount bin/smbmnt bin/smbumount"
+		EXTRA_BIN_PROGS="$EXTRA_BIN_PROGS bin/smbmount"
+		EXTRA_SUBIN_PROGS="$EXTRA_SUBIN_PROGS client/mount.cifs$(EXEEXT) client/umount.cifs$(EXEEXT) bin/smbmnt$(EXEEXT) bin/smbumount$(EXEEXT)"
 		;;
 	*)
 		AC_MSG_ERROR(not on a linux system!)
