diff -ur grub-0.96.orig/stage2/fsys_xfs.c grub-0.96/stage2/fsys_xfs.c
--- grub-0.96.orig/stage2/fsys_xfs.c	2004-01-18 21:18:32.000000000 +0200
+++ grub-0.96/stage2/fsys_xfs.c	2005-12-26 02:20:19.000000000 +0200
@@ -334,7 +334,7 @@
 {
 	int namelen = 1;
 	int toread;
-	static char *usual[2] = {".", ".."};
+	static char usual[2][3] = {{'.', 0, 0}, {'.', '.', 0}};
 	static xfs_dir2_sf_entry_t *sfe;
 	char *name = usual[0];
 
