Index: init.d/checkroot
===================================================================
--- init.d/checkroot	(revision 5740)
+++ init.d/checkroot	(working copy)
@@ -86,10 +86,10 @@
 	fi
     
 	if grep -qs cdroot /proc/cmdline; then
-		sys_lang=`/usr/bin/sed -e 's/.* lang=//' /proc/cmdline|cut -f1 -s -d " "`
+        sys_lang=`cat /proc/cmdline | awk -F"lang=" '{print $2}' | awk '{print $1}'`
         
 		# Fallback/default is Turkish, but we also support English
-		if /usr/bin/test $sys_lang = "en"; then
+        if [[ ${sys_lang} = "en" ]]; then
 			echo "LANG=\"en_US.UTF-8\"" > /etc/env.d/03locale
 			echo "LC_ALL=\"en_US.UTF-8\"" >> /etc/env.d/03locale
 		else
Index: sbin/functions.sh
===================================================================
--- sbin/functions.sh	(revision 5687)
+++ sbin/functions.sh	(working copy)
@@ -176,8 +176,8 @@
 
 # Wrapper function for i18n support
 translate() {
-    sys_lang=`/usr/bin/sed -e 's/.* lang=//' /proc/cmdline|cut -f1 -s -d " "`
-    if /usr/bin/test $sys_lang = "en"; then
+    sys_lang=`cat /proc/cmdline | awk -F"lang=" '{print $2}' | awk '{print $1}'`
+    if [[ ${sys_lang} == "en" ]]; then
         sys_locale="en"
     else
         sys_locale="tr"
@@ -345,7 +345,13 @@
 
 	if [[ ${retval} == 0 ]]; then
 		[[ ${RC_QUIET_STDOUT} == yes ]] && return 0
-		msg="${BRACKET}[ ${GOOD}ok${BRACKET} ]${NORMAL}"
+
+        sys_lang=`cat /proc/cmdline | awk -F"lang=" '{print $2}' | awk '{print $1}'`
+        if [[ ${sys_lang} = "en" ]]; then
+    		msg="${BRACKET}[ ${GOOD}ok${BRACKET} ]${NORMAL}"
+        else
+            msg="${BRACKET}[ ${GOOD}tamam${BRACKET} ]${NORMAL}"
+        fi
 	else
 		if [[ -c /dev/null ]]; then
 			rc_splash "stop" &>/dev/null &
@@ -828,10 +834,16 @@
 	(( COLS > 0 )) || (( COLS = 80 ))	# width of [ ok ] == 7
 fi
 
+sys_lang=`cat /proc/cmdline | awk -F"lang=" '{print $2}' | awk '{print $1}'`
+
 if [[ ${RC_ENDCOL} == yes ]]; then
-	ENDCOL=$'\e[A\e['$(( COLS - 7 ))'G'
+    if [[ ${sys_lang} == "en" ]]; then
+	    ENDCOL=$'\e[A\e['$(( COLS - 7 ))'G'
+    else
+        ENDCOL=$'\e[A\e['$(( COLS - 10 ))'G'
+    fi
 else
-	ENDCOL=''
+    ENDCOL=''
 fi
 
 # Setup the colors so our messages all look pretty
Index: sbin/rc
===================================================================
--- sbin/rc	(revision 5712)
+++ sbin/rc	(working copy)
@@ -162,11 +162,11 @@
 		unset mntcmd
 	fi
 	try mount -n ${mntcmd:--t proc proc /proc}
-
- 	sys_lang=`/usr/bin/sed -e 's/.* lang=//' /proc/cmdline|cut -f1 -s -d " "`
     
+    sys_lang=`cat /proc/cmdline | awk -F"lang=" '{print $2}' | awk '{print $1}'`
+    
 	# Fallback/default is Turkish, but we also support English
- 	if /usr/bin/test $sys_lang = "en"; then
+    if [[ ${sys_lang} = "en" ]]; then
  		sys_keymap="us"
  		sys_font="iso01.16"
  		sys_trans="8859-1"
