diff -Nur mplayer-20051121-old/configure mplayer-20051121/configure
--- mplayer-20051121-old/configure	2005-11-17 13:07:29.000000000 +0200
+++ mplayer-20051121/configure	2005-11-21 15:58:04.000000000 +0200
@@ -5468,6 +5468,7 @@
 echocheck "OggTheora support"
 if test "$_theora" = auto ; then
   _theora=no
+  if ( pkg-config --exists 'theora' ) >> "$TMPLOG" 2>&1 ; then
     cat > $TMPC << EOF
 #include <theora/theora.h>
 #include <string.h>
@@ -5502,22 +5503,16 @@
   return 0;
 }
 EOF
-  for _ld_theora in "`pkg-config --silence-errors --libs --cflags theora`" "-ltheora"; do
-    cc_check $_ld_theora && _theora=yes && break
-  done
-  if test "$_theora" = no && test "$_tremor_internal" = yes; then
-    for _ld_theora in "`pkg-config --silence-errors --libs --cflags theora`" "-ltheora"; do
-      cc_check -I. tremor/bitwise.c $_ld_theora && _theora=yes && break
-    done
+  cc_check `pkg-config --libs --cflags theora` && _theora=yes
   fi
 fi
 if test "$_theora" = yes ; then
   _def_theora='#define HAVE_OGGTHEORA 1'
   _codecmodules="libtheora $_codecmodules"
+  _ld_theora=`pkg-config --libs --cflags theora`
 else
   _def_theora='#undef HAVE_OGGTHEORA'
   _nocodecmodules="libtheora $_nocodecmodules"
-  _ld_theora=""
 fi
 echores "$_theora"
 
