Index: kdecore/klibloader.cpp
===================================================================
--- kdecore/klibloader.cpp	(revision 482753)
+++ kdecore/klibloader.cpp	(working copy)
@@ -181,7 +181,7 @@
     void* sym = lt_dlsym( (lt_dlhandle) m_handle, symname );
     if ( !sym )
     {
-        KLibLoader::self()->d->errorMessage = "KLibrary: " + QString::fromLatin1( lt_dlerror() );
+        KLibLoader::self()->d->errorMessage = "KLibrary: " + QString::fromLocal8Bit( lt_dlerror() );
         kdWarning(150) << KLibLoader::self()->d->errorMessage << endl;
         return 0;
     }
@@ -431,7 +431,7 @@
       {
         const char* errmsg = lt_dlerror();
         if(errmsg)
-            d->errorMessage = QString::fromLatin1(errmsg);
+            d->errorMessage = QString::fromLocal8Bit(errmsg);
         else
             d->errorMessage = QString::null;
         return 0;
