diff -Nur mc-4.6.0-old/lib/mc-wrapper.sh.in mc-4.6.0/lib/mc-wrapper.sh.in
--- mc-4.6.0-old/lib/mc-wrapper.sh.in	2005-12-22 21:12:13.000000000 +0200
+++ mc-4.6.0/lib/mc-wrapper.sh.in	2005-12-22 21:21:43.000000000 +0200
@@ -11,3 +11,14 @@
 
 rm -f "$MC_PWD_FILE"
 unset MC_PWD_FILE
+
+MC ()
+{
+	mkdir -p $HOME/.mc/tmp 2> /dev/null
+	chmod 700 $HOME/.mc/tmp
+	MC=$HOME/.mc/tmp/mc-$$
+	@bindir@/mc -P "$MC"
+	cd "`cat $MC`"
+	rm -f "$MC"
+	unset MC;
+}
