--- x264-20051016/x264.c	2005-10-16 12:22:19.000000000 +0000
+++ x264-work/x264.c	2005-10-16 17:07:53.000000000 +0000
@@ -1412,7 +1412,7 @@
         return -1;
 
     memset(p_mp4, 0, sizeof(mp4_t));
-    p_mp4->p_file = gf_isom_open(psz_filename, GF_ISOM_OPEN_WRITE, NULL);
+    p_mp4->p_file = gf_isom_open(psz_filename, GF_ISOM_OPEN_WRITE);
 
     if ((p_mp4->p_sample = gf_isom_sample_new()) == NULL)
     {
