Index: configure.ac
===================================================================
--- configure.ac
+++ configure.ac
@@ -1112,12 +1112,16 @@ dnl ------------------------------------
 dnl External version of a52dec
 dnl ---------------------------------------------
 
+AC_ARG_ENABLE(a52dec, AC_HELP_STRING([--disable-a52dec], [Disable support for a52dec decoding library (default: enabled)]),
+              [enable_a52dec="$enableval"], [enable_a52dec="yes"])
 AC_ARG_WITH(external-a52dec, AC_HELP_STRING([--with-external-a52dec], [use external a52dec library (not recommended)]),
             [external_a52dec="$withval"], [external_a52dec="no"])
 
 have_a52="no"
 
-if test x"$external_a52dec" = "xyes"; then
+if test x"$enable_a52dec" = "xno"; then
+  AC_MSG_RESULT([a52dec support disabled])
+elif test x"$external_a52dec" = "xyes"; then
   have_a52="yes"
   AC_CHECK_HEADERS([a52dec/a52.h a52dec/a52_internal.h],, have_a52="no",
 [
@@ -1144,17 +1148,23 @@ else
   AC_MSG_RESULT([Use included a52dec support])
 fi
 
+AM_CONDITIONAL(A52, test x"$enable_a52dec" = "xyes")
 AM_CONDITIONAL(EXTERNAL_A52DEC, test x"$have_a52" = "xyes")
 
 dnl ---------------------------------------------
 dnl External version of libmad
 dnl ---------------------------------------------
 
+AC_ARG_ENABLE(mad, AC_HELP_STRING([--disable-mad], [Disable support for MAD decoding library (default: enabled)]),
+              [enable_libmad="$enableval"], [enable_libmad="yes"])
 AC_ARG_WITH(external-libmad, AC_HELP_STRING([--with-external-libmad], [use external libmad library (not recommended)]),
             [external_libmad="$withval"], [external_libmad="no"])
 
 have_mad="no"
-if test x"$external_libmad" = "xyes"; then
+
+if test "x$enable_libmad" = "xno"; then
+  AC_MSG_RESULT([libmad support disabled])
+elif test x"$external_libmad" = "xyes"; then
   PKG_CHECK_MODULES(LIBMAD, [mad], have_mad=yes, have_mad=no)
   AC_CHECK_HEADERS([mad.h])
   AC_SUBST(LIBMAD_LIBS)
@@ -1166,6 +1176,7 @@ else
   AC_MSG_RESULT([Use included libmad support])
 fi
 
+AM_CONDITIONAL(MAD, test "x$enable_libmad" = "xyes")
 AM_CONDITIONAL(EXTERNAL_LIBMAD, test x"$have_mad" = "xyes")
 
 dnl ---------------------------------------------
@@ -1695,6 +1706,33 @@ AC_ARG_ENABLE(faad, AC_HELP_STRING([--di
 	      enable_faad=$enableval, enable_faad=yes)
 AM_CONDITIONAL(BUILD_FAAD, test x"$enable_faad" = "xyes")
 
+dnl ---------------------------------------------
+dnl Optional and external libdts
+dnl ---------------------------------------------
+
+AC_ARG_ENABLE(dts, AC_HELP_STRING([--disable-dts], [Disable support for DTS decoding library (default: enabled)]),
+              [enable_libdts="$enableval"], [enable_libdts="yes"])
+AC_ARG_WITH(external-libdts, AC_HELP_STRING([--with-external-libdts], [use external libdts library (not recommended)]),
+            [external_libdts="$withval"], [external_libdts="no"])
+
+have_dts="no"
+
+if test "x$enable_libdts" = "xno"; then
+  AC_MSG_RESULT([libdts support disabled])
+elif test x"$external_libdts" = "xyes"; then
+  PKG_CHECK_MODULES(LIBDTS, [libdts], have_dts=yes, have_dts=no)
+  AC_CHECK_HEADERS([dts.h])
+  AC_SUBST(LIBDTS_LIBS)
+  AC_SUBST(LIBDTS_CFLAGS)
+  if test "x$have_dts" = "xno"; then
+    AC_MSG_RESULT([*** no usable version of libdts found, using internal copy ***])
+  fi
+else
+  AC_MSG_RESULT([Use included libdts support])
+fi
+
+AM_CONDITIONAL(DTS, test "x$enable_libdts" = "xyes")
+AM_CONDITIONAL(EXTERNAL_LIBDTS, test x"$have_dts" = "xyes")
 
 dnl ---------------------------------------------
 dnl libmodplug support 
@@ -2615,10 +2653,12 @@ fi
 if test x"$have_modplug" = x"yes"; then
   echo "   - mod"
 fi
-if test x"$have_a52" = x"yes"; then
-  echo "   - ac3 (external library)"
-else
-  echo "   - ac3 (internal library)"
+if test x"$enable_a52dec" = "xyes"; then
+  if test x"$have_a52" = "xyes"; then
+    echo "   - ac3 (external library)"
+  else
+    echo "   - ac3 (internal library)"
+  fi
 fi
 echo ""
 
@@ -2661,8 +2701,7 @@ echo ""
 
 dnl audio decoders
 echo " * audio decoder plugins:"
-echo "   - A52/ra-dnet     - DTS"
-echo "   - MAD (MPG 1/2/3) - GSM 06.10"
+echo "   - GSM 06.10"
 echo "   - linear PCM      - Nosefart (NSF)"
 if test x"$external_ffmpeg_found" = "xyes"; then
   echo "   - ffmpeg (external library):"
@@ -2690,6 +2729,27 @@ fi
 if test x"$enable_faad" = "xyes"; then
   echo "   - faad"
 fi
+if test x"$enable_libmad" = "xyes"; then
+  if test x"$have_mad" = "xyes"; then
+    echo "   - MAD (MPG 1/2/3) (external library)"
+  else
+    echo "   - MAD (MPG 1/2/3) (internal library)"
+  fi
+fi
+if test x"$enable_libdts" = "xyes"; then
+  if test x"$have_dts" = "xyes"; then
+    echo "   - DTS (external library)"
+  else
+    echo "   - DTS (internal library)"
+  fi
+fi
+if test x"$enable_a52dec" = "xyes"; then
+  if test x"$have_a52" = "xyes"; then
+    echo "   - A52/ra-dnet (external library)"
+  else
+    echo "   - A52/ra-dnet (internal library)"
+  fi
+fi
 echo ""
 
 dnl spu decoders
Index: src/liba52/Makefile.am
===================================================================
--- src/liba52/Makefile.am
+++ src/liba52/Makefile.am
@@ -2,7 +2,11 @@ include $(top_srcdir)/misc/Makefile.comm
 
 libdir = $(XINE_PLUGINDIR)
 
-lib_LTLIBRARIES = xineplug_decode_a52.la
+if A52
+a52_module = xineplug_decode_a52.la
+endif
+
+lib_LTLIBRARIES = $(a52_module)
 
 if EXTERNAL_A52DEC
 internal_sources =
Index: src/libdts/Makefile.am
===================================================================
--- src/libdts/Makefile.am
+++ src/libdts/Makefile.am
@@ -2,11 +2,27 @@ include $(top_srcdir)/misc/Makefile.comm
 
 libdir = $(XINE_PLUGINDIR)
 
-lib_LTLIBRARIES = xineplug_decode_dts.la
+if DTS
+dts_module = xineplug_decode_dts.la
+endif
 
-xineplug_decode_dts_la_SOURCES = bitstream.c downmix.c parse.c xine_decoder.c
-xineplug_decode_dts_la_LIBADD = $(XINE_LIB)
+lib_LTLIBRARIES = $(dts_module)
+
+if EXTERNAL_LIBDTS
+internal_sources =
+else
+internal_sources = bitstream.c downmix.c parse.c 
+endif
+
+xineplug_decode_dts_la_SOURCES = xine_decoder.c $(internal_sources)
 xineplug_decode_dts_la_LDFLAGS = -avoid-version -module $(XINE_PLUGIN_MIN_SYMS)
 
+if EXTERNAL_LIBDTS
+xineplug_decode_dts_la_LIBADD = $(XINE_LIB) $(LIBDTS_LIBS)
+xineplug_decode_dts_la_CFLAGS = $(LIBDTS_CFLAGS)
+else
+xineplug_decode_dts_la_LIBADD = $(XINE_LIB)
+endif
+
 noinst_HEADERS = bitstream.h dts.h dts_internal.h tables.h tables_adpcm.h \
 	tables_fir.h tables_huffman.h tables_quantization.h tables_vq.h
Index: src/libmad/Makefile.am
===================================================================
--- src/libmad/Makefile.am
+++ src/libmad/Makefile.am
@@ -6,7 +6,11 @@ EXTRA_DIST = imdct_l_arm.S
 
 libdir = $(XINE_PLUGINDIR)
 
-lib_LTLIBRARIES = xineplug_decode_mad.la
+if MAD
+mad_module = xineplug_decode_mad.la
+endif
+
+lib_LTLIBRARIES = $(mad_module)
 
 if EXTERNAL_LIBMAD
 internal_sources =
