Index: kpdf/part.rc
===================================================================
--- kpdf/part.rc	(revision 487785)
+++ kpdf/part.rc	(working copy)
@@ -35,7 +35,7 @@
     <Separator/>
     <Action name="goto_page"/>
   </Menu>
-  <Menu name="mouseMode"><text>&amp;Tools</text>
+  <Menu name="tools"><text>&amp;Tools</text>
     <Action name="mouse_drag"/>
     <Action name="mouse_zoom"/>
     <Action name="mouse_select"/>
