--- binutils-2.12.1/libiberty/fibheap.c.mps	Thu Aug  1 08:49:52 2002
+++ binutils-2.12.1/libiberty/fibheap.c	Thu Aug  1 08:51:07 2002
@@ -34,6 +34,21 @@
 #include "libiberty.h"
 #include "fibheap.h"
 
+/* FIXME: It'd be nice to configure around these, but the include files are too
+   painful.  These macros should at least be more portable than hardwired hex
+   constants. */
+
+#ifndef ULONG_MAX
+#define	ULONG_MAX	((unsigned long)(~0L))		/* 0xFFFFFFFF */
+#endif
+
+#ifndef LONG_MAX
+#define	LONG_MAX	((long)(ULONG_MAX >> 1))	/* 0x7FFFFFFF */
+#endif
+
+#ifndef LONG_MIN
+#define	LONG_MIN	((long)(~LONG_MAX))		/* 0x80000000 */
+#endif
 
 #define FIBHEAPKEY_MIN	LONG_MIN
 
