diff -pruN gtk-qt-engine-old/src/qt_theme_draw.c gtk-qt-engine/src/qt_theme_draw.c
--- gtk-qt-engine-old/src/qt_theme_draw.c	2005-01-06 21:38:14.000000000 +0100
+++ gtk-qt-engine/src/qt_theme_draw.c	2005-08-10 23:28:34.875969476 +0200
@@ -1831,6 +1831,7 @@ set_background (GtkStyle *style, GdkWind
           parent_relative = FALSE;
         }
       
+      if (pixmap && !gdk_drawable_get_colormap (pixmap)) gdk_drawable_set_colormap (pixmap, gdk_drawable_get_colormap (window));
       gdk_window_set_back_pixmap (window, pixmap, parent_relative);
     }
   else
