diff -ur kdebase-3.5.0.orig/kcontrol/kicker/extensionInfo.cpp kdebase-3.5.0/kcontrol/kicker/extensionInfo.cpp
--- kdebase-3.5.0.orig/kcontrol/kicker/extensionInfo.cpp	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kcontrol/kicker/extensionInfo.cpp	2005-11-30 17:05:53.000000000 +0200
@@ -49,7 +49,7 @@
         _customSizeMax = 256;
         _customSize = 56;
         _showLeftHB     = false;
-        _showRightHB    = true;
+        _showRightHB    = false;
 	for (int i=0;i<4;i++) _allowedPosition[i]=true;
     }
     else
@@ -201,7 +201,7 @@
     _xineramaScreen = QApplication::desktop()->primaryScreen();
     _size           = 2;
     _showLeftHB     = false;
-    _showRightHB    = true;
+    _showRightHB    = false;
     _hideButtonSize = 14;
     _autohidePanel  = false;
     _backgroundHide = false;
diff -ur kdebase-3.5.0.orig/kcontrol/kicker/hidingtab_impl.cpp kdebase-3.5.0/kcontrol/kicker/hidingtab_impl.cpp
--- kdebase-3.5.0.orig/kcontrol/kicker/hidingtab_impl.cpp	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kcontrol/kicker/hidingtab_impl.cpp	2005-11-30 17:05:53.000000000 +0200
@@ -201,7 +201,7 @@
     m_delaySpinBox->setValue(3);
     m_autoHideSwitch->setChecked(false);
     m_lHB->setChecked( false );
-    m_rHB->setChecked( true );
+    m_rHB->setChecked( false );
     m_animateHiding->setChecked(true);
     m_hideSlider->setValue(10);
     m_delaySpinBox->setValue(3);
diff -ur kdebase-3.5.0.orig/kcontrol/kicker/lookandfeeltab_impl.cpp kdebase-3.5.0/kcontrol/kicker/lookandfeeltab_impl.cpp
--- kdebase-3.5.0.orig/kcontrol/kicker/lookandfeeltab_impl.cpp	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kcontrol/kicker/lookandfeeltab_impl.cpp	2005-11-30 17:05:53.000000000 +0200
@@ -172,13 +172,13 @@
     browserTileChanged(m_browserTile->currentItem());
     wlTileChanged(m_windowListTile->currentItem());
 
-    if (config.readBoolEntry("EnableTileBackground", false))
+    if (config.readBoolEntry("EnableTileBackground", true))
     {
         config.setGroup("button_tiles");
 
-        if (config.readBoolEntry("EnableKMenuTiles", false))
+        if (config.readBoolEntry("EnableKMenuTiles", true))
         {
-            tile = config.readEntry("KMenuTile", "solid_blue");
+            tile = config.readEntry("KMenuTile", "solid_pardus");
             m_kmenuTile->setCurrentItem(m_tilename.findIndex(tile));
             kcfg_KMenuTileColor->setEnabled(tile == "Colorize");
         }
diff -ur kdebase-3.5.0.orig/kcontrol/kicker/menutab_impl.cpp kdebase-3.5.0/kcontrol/kicker/menutab_impl.cpp
--- kdebase-3.5.0.orig/kcontrol/kicker/menutab_impl.cpp	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kcontrol/kicker/menutab_impl.cpp	2005-11-30 17:05:53.000000000 +0200
@@ -114,7 +114,7 @@
         }
     }
 
-    m_showFrequent->setChecked(true);
+    m_showFrequent->setChecked(false);
 }
 
 void MenuTab::save()
@@ -154,10 +154,10 @@
     {
          static_cast<kSubMenuItem*>( item )->setOn( false );
     }
-    m_bookmarkMenu->setOn(true);
-    m_quickBrowserMenu->setOn(true);
+    m_bookmarkMenu->setOn(false);
+    m_quickBrowserMenu->setOn(false);
 
-    m_showFrequent->setChecked(true);
+    m_showFrequent->setChecked(false);
 }
 
 void MenuTab::launchMenuEditor()
diff -ur kdebase-3.5.0.orig/kicker/kicker/core/default-apps kdebase-3.5.0/kicker/kicker/core/default-apps
--- kdebase-3.5.0.orig/kicker/kicker/core/default-apps	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kicker/kicker/core/default-apps	2005-11-30 17:41:52.000000000 +0200
@@ -1,2 +1,3 @@
 Home.desktop
-Internet/konqbrowser.desktop
+Internet/Kontact.desktop
+Internet/mozillafirefox.desktop
diff -ur kdebase-3.5.0.orig/kicker/libkicker/kickerSettings.kcfg kdebase-3.5.0/kicker/libkicker/kickerSettings.kcfg
--- kdebase-3.5.0.orig/kicker/libkicker/kickerSettings.kcfg	2005-11-30 17:05:34.000000000 +0200
+++ kdebase-3.5.0/kicker/libkicker/kickerSettings.kcfg	2005-11-30 17:05:53.000000000 +0200
@@ -12,7 +12,7 @@
 
 <entry name="Locked" type="Bool" >
       <label>When this option is enabled, the panels may not be moved and items cannot be removed or added</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="ConserveSpace" type="Bool" >
@@ -128,7 +128,7 @@
 
 <entry name="MenuEntryHeight" type="Int" >
       <label>Height of menu entries in pixels</label>
-      <default>0</default>
+      <default>22</default>
    </entry>
 
 <entry name="ShowHiddenFiles" type="Bool" >
@@ -163,13 +163,13 @@
 
 <entry name="NumVisibleEntries" type="UInt" >
       <label>Number of visible entries</label>
-      <default>5</default>
+      <default>3</default>
       <max>100</max>
    </entry>
 
 <entry name="RecentVsOften" type="Bool" >
       <label>Show most recently used applications rather than most frequently used</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 </group>
@@ -178,7 +178,7 @@
 
 <entry name="EnableKMenuTiles" type="Bool">
       <label>Enable a tile background image for the KMenu button</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="EnableDesktopButtonTiles" type="Bool">
@@ -203,6 +203,7 @@
 
 <entry name="KMenuTile" type="Path" >
       <label>Image tile for Kmenu button background</label>
+      <default>solid_pardus</default>
    </entry>
 
 <entry name="KMenuTileColor" type="Color" >
@@ -267,12 +268,12 @@
 
 <entry name="ShowKMenuText" key="ShowText" type="Bool" >
       <label>Show text on the K Menu button</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="KMenuText" key="Text" type="String" >
       <label>Text to be shown on K Menu Button</label>
-      <default code="true">i18n("Applications")</default>
+      <default code="true">i18n("Pardus")</default>
    </entry>
 
 </group>
@@ -312,7 +313,7 @@
 
 <entry name="EnableTileBackground" type="Bool" >
       <label>Enable background tiles</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="IconMargin" key="IconMargin" type="Int" >
